core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -952,6 +952,12 @@ private[spark] class SparkSubmit extends Logging {
     } catch {
       case t: Throwable =>
         throw findCause(t)
+    } finally {
+      try {
+        SparkContext.getActive.foreach(_.stop())
```

Comment:
Just reading this and https://github.com/apache/spark/pull/33154, shouldn't we enable this only w/ Kubernates (and also when it's not a Thirftserver, shall, etc.)?

Also, I think we might have to add some comments on that.

Reply:
cc @sunpe too FYI

Reply:
@HyukjinKwon This PR was reverted once and resubmitted https://github.com/apache/spark/pull/32283 (in which Thirftserver and others are excluded).

But I agree with you that we should only do this for K8s. Otherwise, it looks like a behavior change to me.

Reply:
Shall we continue the discussion on the latest one, #32282 , please?

Reply:
@dongjoon-hyun that pr looks unrelated, right ? Not sure if I am missing something.

Reply:
Oops. It's a typo of #32283 . I tried to mention the one from @Ngone51 's comment.

Reply:
Thanks @dongjoon-hyun ! Was not sure if there was another pr also I should be looking at :-)